### PR TITLE
feat(cli): Implement `mun new` and `mun init`

### LINKED ISSUE #237: Implement `mun new` and `mun init`
Now that [projects have landed](#223) it would be good if we can create a barebones project by invoking `mun new` or `mun init` similar to [`cargo new`](https://doc.rust-lang.org/cargo/commands/cargo-new.html) and [`cargo init`](https://doc.rust-lang.org/cargo/commands/cargo-init.html).

**Good first issue process:**

- [ ] [read our contributing guidelines](https://github.com/mun-lang/mun/blob/master/CONTRIBUTING.md)
- [ ] claim this issue (assign yourself or comment below)
- [ ] implement the feature and start a Pull Request with in the description: closes #237. 

If this is your first PR, welcome 🎉 😄

Related to #223

### PR DESCRIPTION
**Pull request by ethanboxx**
closes #237 

The `cargo` implementation was used as a reference.

## Note

I did not do anything to fetch information about the project author.

Cargo uses the function https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_new.rs#L773.

I could copy it over almost line for line but that feels a little weird. Maybe I should create a small crate for it. Do you think that would be a good idea?

### CODE REVIEW

**Comment by ethanboxx** [comment]
**Note:** I decided to not initialize git to keep this PR as simple as possible. I will probably add that feature in a later PR.

File: crates/mun/src/lib.rs [hunk 1]
```diff
@@ -278,3 +288,88 @@ mod test {
...
+/// This is used by `init` and `new` arguments to create projects in different paths.
+fn create_project(create_in: &Path, project_name: &str) -> Result<ExitStatus, anyhow::Error> {
+    log::trace!("Creating new project");
+    {
+        let manifest_path = create_in.join("mun.toml");
+
+        write(
+            &manifest_path,
+            format!(
+                // @TODO. Nothing is done yet to find out the author is.
```
Comment (baszalmstra) [inline crates/mun/src/lib.rs:337]:
```suggestion
                // @TODO. Nothing is done yet to find out who the author is.
```

File: crates/mun/src/lib.rs [hunk 2]
```diff
@@ -94,6 +95,13 @@ where
                         .help("how much to delay received filesystem events (in ms). This allows bundling of identical events, e.g. when several writes to the same file are detected. A high delay will make hot reloading less responsive. (defaults to 10 ms)"),
                 ),
         )
+        .subcommand(
+            SubCommand::with_name("new")
+                .arg(Arg::with_name("path").help("the path to crate a new project").required(true).index(1))
```
Comment (Wodann) [inline crates/mun/src/lib.rs:100]:
```suggestion
                .arg(Arg::with_name("path").help("the path to create a new project").required(true).index(1))
```

File: crates/mun/src/lib.rs [hunk 3]
```diff
@@ -278,3 +288,88 @@ mod test {
...
+    );
+
+    let create_in = std::env::current_dir()
+        .expect("could not determine current working directory")
+        .join(project_name);
+
+    if create_in.exists() {
+        eprint!(
+            "destination `{}` already exists\n\n\
+             Use `mun init` to initialize the directory",
```
Comment (Wodann) [inline crates/mun/src/lib.rs:306]: ❤️ the error message's UI/UX

**Comment by ethanboxx** [comment]
Done.

File: crates/mun/tests/integration.rs [hunk 4]
```diff
@@ -1,63 +1,57 @@
...
+        .unwrap();
+
+    set_current_dir(&project).unwrap();
+
+    let args: Vec<OsString> = vec!["mun".into(), "init".into()];
+    assert_eq!(run_with_args(args).unwrap(), mun::ExitStatus::Success);
+    build_and_run(project);
 }
 
+/// Crates a new project using `mun new` and then tests that it works.
```
Comment (Wodann) [inline crates/mun/tests/integration.rs:24]:
```suggestion
/// Creates a new project using `mun new` and then tests that it works.
```

**Comment by baszalmstra** [comment]
Yes! Looks good to me as well, if you can clean up your history a little bit, Ill merge this. :)

**Comment by ethanboxx** [comment]
Done

**Comment by Wodann** [comment]
Thanks for rebasing. A much cleaner history now; very pleasing 🙂

**Comment by baszalmstra** [comment]
Relates to #203

**Pull request merged**